### ISSUE TITLE
Folding incorrect

### ISSUE THREAD

**Issue by DarkLite1** [PowerShell/vscode-powershell#1313]
### System Details

- Operating system name and version: Windows Server 2012R2
- VS Code version: 1.24-0-insider
- PowerShell extension version: 1.7.0
- Output from 

```
Name                           Value
----                           -----
PSVersion                      5.1.14409.1012
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.14409.1012
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1:
```

### Issue Description

I am experiencing a problem with the folding of code. Now that the new API allows language aware folding, it would be nice if this could be folded correctly:

```
$ThresholdTest = @(
    @{
        Percentage = 85
        Color      = 'Red'
        Action     = @{
            MailTo           = 'bob@contoso.com'
            Subject          = "Your personal home drive has reached [Quota Threshold]% of its maximum allowed volume"
            Body             = @"
Dear [Source Io Owner],

Your personal home drive has reached [Quota Threshold]% of its 
"@
            Type             = 2
            RunLimitInterval = 2880
        }
    },
    @{
        Percentage = 100
        Color      = 'Green'
        Action     = @{
            MailTo           = 'bob@contoso.com'
            Subject          = 'Your personal home drive has reached or exceeded its maximum allowed volume'
            Body             = @"
Dear [Source Io Owner],

Your personal home drive has now reached [Quota Threshold]% of its maximum allowed volume.
"@
            Type             = 2
            RunLimitInterval = 2880
        }
    }
)
```

When clicking `CTRL+K CTRL+O` it looks like this:
![image](https://user-images.githubusercontent.com/10828570/39690900-f349b6b8-51db-11e8-9b25-28dc9fd876db.png)

**Comment by rkeithhill** [comment]
Duplicate of #177.  The plan is to add this support. Keep in mind the VSCode API to support this - just shipped in 1.23.

**Comment by DarkLite1** [comment]
Awesome, thank you @rkeithhill
